core/src/main/kotlin/org/evomaster/core/problem/rest/service/ResourceSampleMethodController.kt

```diff
@@ -0,0 +1,337 @@
+package org.evomaster.core.problem.rest.service
+
+import com.google.inject.Inject
+import org.evomaster.core.EMConfig
+import org.evomaster.core.problem.rest.resource.RestResourceNode
+import org.evomaster.core.problem.rest.resource.SamplerSpecification
+import org.evomaster.core.problem.rest.service.ResourceSamplingMethod.*
+import org.evomaster.core.search.service.Randomness
+import org.evomaster.core.search.service.SearchTimeController
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+/**
+ * dynamically determinate resource-based sample method
+ */
+class ResourceSampleMethodController {
+
+    @Inject
+    private lateinit var time : SearchTimeController
+
+    @Inject
+    private lateinit var randomness: Randomness
+
+    @Inject
+    private lateinit var config : EMConfig
+
+    @Inject
+    private lateinit var rm : ResourceManageService
+
+    private val methods : Map<ResourceSamplingMethod, MethodApplicationInfo> = values().map { Pair(it, MethodApplicationInfo()) }.toMap()
+
+    companion object {
+        private val log: Logger = LoggerFactory.getLogger(ResourceSampleMethodController::class.java)
+        private const val CONARCHIVE_THRESHOLD = 0.2
+        private const val TB_THRESHOLD = 0.5
+
+    }
+
+    fun initialize(){
+        initApplicableStrategies()
+        initProbability()
+        validateProbability()
+    }
+
+    private fun initApplicableStrategies(){
+        methods.getValue(S1iR).applicable = true //mutableMap.values.filter { r -> r.hasIndependentAction }.isNotEmpty()
+        rm.getResourceCluster().values.filter { r -> !r.isIndependent() }.let {
+            methods.getValue(S1dR).applicable = it.isNotEmpty()
+            methods.getValue(S2dR).applicable = it.size > 1 && config.maxTestSize > 1
+            methods.getValue(SMdR).applicable = it.size > 2 && config.maxTestSize > 2
+        }
+
+        //FIXME Man Zhang
+        if(methods.values.filter { it.applicable }.size == 1 ) config.maxTestSize = 1
+    }
+
+    private fun printApplicableStr(){
+        println("Applicable SmartSampleStrategy>>>")
+        println( methods.filter{ it.value.applicable }.mapNotNull { "${it.key} : ${it.value.probability}"}.joinToString (" - "))
+    }
+
+    private fun printSummaryOfResources(mutableMap: Map<String, RestResourceNode>){
+        println("Summary of abstract resources and actions>>>")
```

Comment:
is this `println` (and the others in this file) actual outputs on standard output? if so, should never be done, and rather use something like `log.info` or `log.debug`

Reply:
no. It is just enabled it when I am debugging. I have modified the code.